src/misc/mixing.c

```diff
@@ -414,3 +414,141 @@ igraph_error_t igraph_assortativity_degree(const igraph_t *graph,
 
     return IGRAPH_SUCCESS;
 }
+
+/**
+ * \function igraph_jdm
+ * \brief The joint degree matrix of a graph.
+ *
+ * \experimental
+ *
+ * This function constructs the joint degree matrix of the graph \p jdm.
+ * <code>jdm[i-1, j-1]</code> will contain the number of edges (or sum of edge
+ * weights) between vertices of degrees <code>i-1</code> and <code>j-1</code>.
```

Comment:
Shouldn’t this be i and j here instead of i-1 and j-1?

Reply:
No. This is a deliberate choice. We omit the row and column corresponding to zero degrees, as they are always all-zero, and not useful. If we use 1-based indexing, then $J_{ij}$ is indeed the number of connections between vertices of degree $i$ and $j$.

Reply:
Okay, but that's not what you wrote there. Substituting `i = 1` and `j = 1`, the sentence reads as "jdm[0, 0] will contain the number of edges (or sum of edge weights) between vertices of degrees 0 and 0".

Initially I wanted to suggest that the end of the sentence should read as "vertices of degrees i and j", but considering that even we got confused by the change in indexing, I think we should just keep everything 0-based for sake of consistency and accept that the first column and the first row of the matrix will be zero by definition. We only waste O(n) space with it, while the size of the matrix is O(n^2) so it's negligible as n goes to infinity.

Reply:
Sorry about that, I fixed is to say "vertices of degrees i and j".

The literature on the subject generally defines the JDM _without_ the zero-degree row/column. Adding that row/column for indexing reasons would give a preference to zero-based indexing languages, but igraph is also used from 1-based indexing ones. We have other functions with similar issues where the first entry corresponds to degree 1, e.g. the `knnk` output parameter of `igraph_avg_nearest_neighbor_degree()`. For these reasons, I would rather omit the first column.

Reply:
Good point on the index base being language-dependent. Also, if the JDM is defined this way in the literature, let's be consistent with the literature. Thanks for fixing the docs, this is now good to go.